crates/bevy_color/src/color_ops.rs

```diff
@@ -0,0 +1,43 @@
+/// Methods for changing the luminance of a color. Note that these methods are not
+/// guaranteed to produce the same results in different color spaces, but are only
+/// approximations.
+pub trait Luminance: Sized {
+    /// Return the luminance of this color (0.0 - 1.0).
+    fn luminance(&self) -> f32;
+
+    /// Return a new version of this color with the given luminance.
+    fn with_luminance(&self, alpha: f32) -> Self;
+
+    /// Return a darker version of this color. The `amount` should be between 0.0 and 1.0.
+    /// The amount represents an absolute decrease in luminance, and is commutative:
+    /// `color.darker(a).darker(b) == color.darker(a + b)`.
+    ///
+    /// For a relative decrease in luminance, you can simply `mix()` with black.
+    fn darker(&self, amount: f32) -> Self;
+
+    /// Return a lighter version of this color. The `amount` should be between 0.0 and 1.0.
+    /// The amount represents an absolute increase in luminance, and is commutative:
+    /// `color.lighter(a).lighter(b) == color.lighter(a + b)`.
```

Comment:
This is a distributive property, not commutative.

Reply:
I don't think distributivity is fully correct either. Perhaps it's best to just leave out the terminology?

Reply:
I believe the correct term is "linear", since the property is `f(a + b) = f(a) + f(b)`, which is a specialised case of the definition of a linear operation `f(ax + b) = a*f(x) + f(b)`

Reply:
At the risk of dangerous levels of bikeshedding... Rewriting `lighter` to $\ell$ you might phrase it as $\ell(b) \circ \ell(a) = \ell(a + b)$ which could be called something like additivity or additivity under composition.